File: 백준/Silver/10845. 큐/큐.py
from collections import deque
q = deque()

input = __import__('sys').stdin.readline

N = int(input())

for i in range(N):
    t = list(input().split())

    if t[0] == 'push':
        q.append(int(t[1]))
    elif t[0] == 'pop':
        if len(q) != 0:
            print(q.popleft())
        else:
            print(-1)
    elif t[0] == 'size':
        print(len(q))
    elif t[0] == 'empty':
        if len(q) == 0:
            print(1)
        else:
            print(0)
    elif t[0] == 'front':
        if len(q) != 0:
            print(q[0])
        else:
            print(-1)
    elif t[0] == 'back':
        if len(q) != 0:
            print(q[-1])
        else:
            print(-1)

File: 백준/Silver/10845. 큐/큐.cc
#include <iostream>
#include <queue>
#include <string>
using namespace std;

int main() {
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    queue<int> q;
    
    string str;
    int n;

    int N;
    cin >> N;
    for (int i=0; i<N; i++) {
        cin >> str;

        if (str == "push") {
            cin >> n;
            q.push(n);
        } else if (str == "pop") {
            if (!q.empty()) {
                cout << q.front() << '\n';
                q.pop();
            } else {
                cout << -1 << '\n';
            }
        } else if (str == "size") {
            cout << q.size() << '\n';
        } else if (str == "empty") {
            cout << q.empty() << '\n';
        } else if (str == "front") {
            if (!q.empty()) {
                cout << q.front() << '\n';
            } else {
                cout << -1 << '\n';
            }
        } else if (str == "back") {
            if (!q.empty()) {
                cout << q.back() << '\n';
            } else {
                cout << -1 << '\n';
            }
        }
    }
    return 0;
}